modules/openapi-generator-cli/pom.xml

```diff
@@ -91,9 +96,9 @@
             <version>2.3.3</version>
         </dependency>
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-simple</artifactId>
-            <version>${slf4j-version}</version>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.0.13</version>
```

Comment:
Is there a reason why you have picked version `1.0.13` for `logback-classic`?

It seems to me that this is an implementation for `org.slf4j:slf4j-api:1.7.5` maybe it is compatible with the version `1.7.12` that we have defined (I guess also without any good reason)

Reply:
I used the version documented as the native implementation on slf4j: https://www.slf4j.org/manual.html

Reply:
This page seems outdated to me... See: https://logback.qos.ch/download.html

Or on maven central:
http://mvnrepository.com/artifact/ch.qos.logback/logback-classic

Do we want to update to:
`logback-classic` version `1.2.3` and `slf4j` to it corresponding version `1.7.25`. This way we will be in sync with `swagger-core`.

Reply:
@jmini this won't necessarily work. SLF4J's site is current.

Let me explain how slf4j works, as there seems to be some confusion.

slf4j provides an abstraction atop common logging frameworks. It does this via bindings that just need to exist in your classpath. It natively includes bindings to logback. This is why you don't have to add a binding jar, only the logback implementation. The logback usage is locked to logback 1.0.13 (see https://github.com/qos-ch/slf4j/blob/master/pom.xml) because it uses interfaces specific to that version. If you upgrade to a newer version of logback, you'd have to check if it is ABI compatible with 1.0.13. If it isn't, you can have exceptions thrown when code is JIT'd. That's obviously not ideal. It *might* just work with some luck, but I see no reason to chance it.

Similarly, if we don't exclude the simple logger binding explicitly, it opens us to the possibility of pulling this in as a transitive dependency. This simple logging package is the only one I've found that writes to stderr only, so allowing that to be included would result in a regression in this bug.

Reply:
Thank you a lot for the explanations... `1.0.13` seems to be the version that we need to use.